src/java.base/share/classes/java/lang/CharSequence.java

```diff
@@ -332,7 +331,8 @@ public static int compare(CharSequence cs1, CharSequence cs2) {
      * @throws     NullPointerException if {@code dst} is {@code null}
      *
      * @implSpec
-     * The default implementation iterates over {@link #charAt(int)}.
+     * The default implementation invokes {@link #charAt(int index)} in a loop
+     * iterating {@code index} from {@code srcBegin} to {@code srcEnd-1}
```

Comment:
We might need to specify the IOOBE behavior - when an IOOBE is thrown, some characters may be already transferred (this is important for concurrent char sequences)

Reply:
Good idea. Added clarification in https://github.com/openjdk/jdk/pull/21730/commits/d2ccf42cec148a86b17753d160df211925f75858.